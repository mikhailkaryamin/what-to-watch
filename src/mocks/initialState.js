const initialState = {
  COMMENT: {
    comments: [{
      id: 1,
      user: {
        id: 1,
        name: `Simple name`,
      },
      rating: 9,
      text: `some text`,
      date: 1293843434,
    },
    {
      id: 2,
      user: {
        id: 2,
        name: `Simple name`,
      },
      rating: 6,
      text: `some text`,
      date: 129384343,
    },
    {
      id: 3,
      user: {
        id: 3,
        name: `Simple name`,
      },
      rating: 9,
      text: `some text`,
      date: 12938034343183,
    }],
    statusUploadComment: null,
  },
  FILMS: {
    films: [
      {
        backgroundImage: `img/aviator.jpg`,
        description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
        director: `Arthur Agee`,
        genre: `Comedy`,
        id: 1,
        name: `Devin Albert`,
        posterImage: `img/aviator.jpg`,
        previewImage: `img/aviator.jpg`,
        previewVideoLink: `link`,
        rating: 3,
        released: 2015,
        runTime: 100,
        scoreCount: 300,
        starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
        video: `some/link`,
      },
      {
        backgroundImage: `img/aviator.jpg`,
        description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
        director: `Arthur Agee`,
        genre: `Comedy`,
        id: 2,
        name: `Devin Albert`,
        posterImage: `img/aviator.jpg`,
        previewImage: `img/aviator.jpg`,
        previewVideoLink: `link`,
        rating: 2,
        released: 2015,
        runTime: 100,
        scoreCount: 300,
        starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
        video: `some/link`,
      },
      {
        backgroundImage: `img/aviator.jpg`,
        description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
        director: `Arthur Agee`,
        genre: `Comedy`,
        id: 3,
        name: `Devin Albert`,
        posterImage: `img/aviator.jpg`,
        previewImage: `img/aviator.jpg`,
        previewVideoLink: `link`,
        rating: 3,
        released: 2015,
        runTime: 100,
        scoreCount: 300,
        starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
        video: `some/link`,
      }
    ],
    promoFilm: {
      backgroundImage: `img/aviator.jpg`,
      description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
      Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
      director: `Arthur Agee`,
      genre: `Comedy`,
      id: 1,
      name: `Devin Albert`,
      posterImage: `img/aviator.jpg`,
      previewImage: `img/aviator.jpg`,
      previewVideoLink: `link`,
      rating: 3,
      released: 2015,
      runTime: 100,
      scoreCount: 300,
      starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
      video: `some/link`,
    },
    statusLoad: false,
  },
  OPTIONS: {
    amountRenderFilmCard: 8,
    genre: `All genres`,
    film: null,
  },
  USER: {
    authStatus: `NO_AUTH`,
    user: {
      id: 777,
      email: `mail@hotmail.com`,
      name: `Vasya`,
      avatarUrl: `/img/pic.jpg`,
    },
  }
};


export {initialState};
