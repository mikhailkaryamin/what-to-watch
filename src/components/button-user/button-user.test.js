import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';

import {initialState} from '../../mocks/initialState.js';

import ButtonUser from './button-user.jsx';

const mockStore = configureStore([]);

describe(`button user`, () => {
  const store = mockStore(initialState);

  const wrapper = renderer
    .create(
        <Provider store={store}>
          <Router>
            <ButtonUser />
          </Router>
        </Provider>
    ).toJSON();

  test(`should render button user`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
