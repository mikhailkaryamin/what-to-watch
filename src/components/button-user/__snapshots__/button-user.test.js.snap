// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button user should render button user 1`] = `
<a
  href="/mylist"
  onClick={[Function]}
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="https://htmlacademy-react-3.appspot.com//img/pic.jpg"
      width="63"
    />
  </div>
</a>
`;
